ui/src/app/services/sign-in.service.ts

```diff
@@ -1,87 +1,105 @@
-import {Injectable} from '@angular/core';
+import 'rxjs/Rx';
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {User} from 'app/models/user';
 
 declare const gapi: any;
 
+interface BasicProfile {
```

Comment:
These two interfaces pretty much come from gapi... they may not be exactly what we really want.